In [5]:
import pandas as pd
import re

In [6]:
data = pd.read_csv('thanksgiving.csv',encoding='Latin-1')
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [7]:
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [8]:
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [9]:
data = data[data['Do you celebrate Thanksgiving?'] == 'Yes']
data['Do you celebrate Thanksgiving?'].value_counts()

Yes    980
Name: Do you celebrate Thanksgiving?, dtype: int64

In [10]:
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [11]:
tofurkey_filter = data[data['What is typically the main dish at your Thanksgiving dinner?'] == 'Tofurkey']
tofurkey_filter['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [12]:
tofurkey_filter['Do you typically have gravy?'].value_counts()

Yes    12
No      8
Name: Do you typically have gravy?, dtype: int64

In [13]:
apple_pie = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple']
pecan_pie = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan']
pumpkin_pie = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin']


In [14]:
apple_isnull = pd.isnull(apple_pie)
pumpkin_isnull = pd.isnull(pumpkin_pie)
pecan_isnull = pd.isnull(pecan_pie)

In [15]:
ate_pies = apple_isnull & pumpkin_isnull & pecan_isnull

In [16]:
ate_pies.iloc[:5]

0    False
1    False
2    False
3    False
4    False
dtype: bool

In [17]:
ate_pies.value_counts()

False    876
True     104
dtype: int64

# Findings

1. There are 876 people who either eat pecan, apple or pumpkin        pies and 104 people who eat neither on their thanksgiving.

# Age analysis

In [18]:
def convert_age(age):
    if pd.isnull(age):
        return None
    else:
        return int(re.split('[- +]',age)[0])               

In [19]:
ages = data['Age']
ages[:3]

0    18 - 29
1    18 - 29
2    18 - 29
Name: Age, dtype: object

In [20]:
ages = ages.apply(convert_age)
ages.head()

0    18.0
1    18.0
2    18.0
3    30.0
4    30.0
Name: Age, dtype: float64

In [21]:
data['int_age'] = ages

In [22]:
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

    ...   Have you ever attended a "Friendsgiving?"  \
0   ...                                          No   
1   ...                                          No   
2   ...                                         Yes   
3   ...                                          No   
4   ...                                          No   

  Will you shop any Black Friday sales on Thanksgiving Day?  \


In [23]:
data['int_age'].describe()

count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%       30.000000
50%       45.000000
75%       60.000000
max       60.000000
Name: int_age, dtype: float64

In [24]:
data['Age'].value_counts()

45 - 59    269
60+        258
30 - 44    235
18 - 29    185
Name: Age, dtype: int64

In [25]:
data['int_age'].value_counts()

45.0    269
60.0    258
30.0    235
18.0    185
Name: int_age, dtype: int64

# Findings

Above shows rough estimation about age infos since we took only initial values of age ranges. But we can say age data is evenly distributed.

# Working on Incomes

In [26]:
money_series=data['How much total combined money did all members of your HOUSEHOLD earn last year?']
money_series.value_counts()

$25,000 to $49,999      166
$75,000 to $99,999      127
$50,000 to $74,999      127
Prefer not to answer    118
$100,000 to $124,999    109
$200,000 and up          76
$10,000 to $24,999       60
$0 to $9,999             52
$125,000 to $149,999     48
$150,000 to $174,999     38
$175,000 to $199,999     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

In [27]:
def convert_money(row):
    if pd.isnull(row):
        return None
    else:
        s = row.split(" ")[0]
        if s == "Prefer":
            return None
        else:
            return int(s[1:].replace(",",""))
        

In [28]:
money_series = money_series.apply(convert_money)
money_series.head()

0     75000.0
1     50000.0
2         0.0
3    200000.0
4    100000.0
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: float64

In [29]:
money_series.value_counts()

25000.0     166
75000.0     127
50000.0     127
100000.0    109
200000.0     76
10000.0      60
0.0          52
125000.0     48
150000.0     38
175000.0     26
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: int64

In [30]:
data['int_income'] = money_series

In [31]:
money_series.describe()

count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%       25000.000000
50%       75000.000000
75%      100000.000000
max      200000.000000
Name: How much total combined money did all members of your HOUSEHOLD earn last year?, dtype: float64

# Corelating distance and income

In [37]:
data[data['int_income'] < 75000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         147
Thanksgiving is local--it will take place in the town I live in                     138
Thanksgiving is out of town but not too far--it's a drive of a few hours or less     94
Thanksgiving is out of town and far away--I have to drive several hours or fly       26
Name: How far will you travel for Thanksgiving?, dtype: int64

In [38]:
data[data['int_income'] > 75000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         142
Thanksgiving is local--it will take place in the town I live in                      74
Thanksgiving is out of town but not too far--it's a drive of a few hours or less     49
Thanksgiving is out of town and far away--I have to drive several hours or fly       32
Name: How far will you travel for Thanksgiving?, dtype: int64

# Linking Friendsgiving,Thanksgiving,Age and Income

In [34]:
data.pivot_table(index = "Have you ever tried to meet up with hometown friends on Thanksgiving night?",
                columns = 'Have you ever attended a "Friendsgiving?"',
                values = "int_age")

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [35]:
data.pivot_table(index = "Have you ever tried to meet up with hometown friends on Thanksgiving night?",
                columns = 'Have you ever attended a "Friendsgiving?"',
                values = "int_income")

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842